src/main/java/sparksoniq/spark/iterator/flowr/FlworExpressionSparkRuntimeIterator.java

```diff
@@ -67,6 +69,39 @@ public FlworExpressionSparkRuntimeIterator(FlowrClauseSparkIterator startClause,
         this._clauses.forEach(clause -> {
                 clause.setDynamicContext(_currentDynamicContext);
         });
+
+        currentResultIndex = 0;
+        this._rdd = this.getRDD();
+        if(SparkContextManager.LIMIT_COLLECT()) {
+            result = _rdd.take(SparkContextManager.COLLECT_ITEM_LIMIT);
+        }
+        else {
+            result = _rdd.collect();
+        }
+
+        if (result.size() == 0) {
+            this._hasNext = false;
+        } else {
+            this._hasNext = true;
+        }
+    }
+
+    @Override
+    public Item next(){
+        if(!this._isOpen) {
+            throw new IteratorFlowException("Runtime iterator is not open", getMetadata());
+        }
+
+        if (this.hasNext()) {
```

Comment:
As a general rule, next() does not need to call hasNext(): it can simply look at the _hasNext field directly (assuming hasNext() consistently returns _hasNext). This avoids an extra call.